domain/secret/errors/errors.go

```diff
@@ -28,4 +28,12 @@ const (
 
 	// AutoPruneNotSupported describes an error that occurs when a charm secret tries to set auto prune on a secret.
 	AutoPruneNotSupported = errors.ConstError("charm secrets do not support auto prune")
+
+	// InvalidSecretPermissionChange describes an error that occurs when an attempt is made to update a secret permission
```

Comment:
Have we put these errors in the apierrors.go?

Reply:
No need to as they are not specifically used by the api caller. If they ever do get used by the api caller, we can add them then.